packages/admin-ui/src/table/components/table-cell.tsx

```diff
@@ -2,23 +2,54 @@ import { createComponent, useElement } from '@vtex/admin-ui-react'
 import type { VariantProps } from '@vtex/admin-ui-core'
 import * as styles from '../styles/table-cell.styles'
 import type { TableColumn } from '../types'
+import React, { useRef, useEffect } from 'react'
+import { useStateContext } from '../context'
+import { Box } from '../../box'
 
 export const TableCell = createComponent<'div', CellOptions>((props) => {
-  const { column, onClick, role = 'cell', ...cellProps } = props
+  const { column, onClick, role = 'cell', ref, children, ...cellProps } = props
+  const { columns } = useStateContext()
 
   const clickable = !!onClick
 
+  const isFixed = !!column?.fixed
+
+  const lastFixedColumn = columns.filter((col) => !!col?.fixed).slice(-1)[0]
+
+  const belongsToLastFixedColumn = isFixed && lastFixedColumn === column
```

Comment:
Just a suggestion to improve the readability, feel free to accept or not. IMO it's not necessary to check if it's fixed twice, the column id will guarantee this to us

```suggestion
  const { id: currentId, fixed: isFixed = false } = column

  const [lastFixedColumn] = columns.filter((col) => !!col?.fixed).slice(-1)

  const isLastFixedColumn = currentId === lastFixedColumn.id
```